internal/k8s/watch.go

```diff
@@ -2,23 +2,59 @@ package k8s
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/pkg/errors"
 	"github.com/windmilleng/tilt/internal/model"
 
 	"k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/watch"
 )
 
-func (kCli K8sClient) WatchPods(ctx context.Context, ls labels.Selector) (<-chan *v1.Pod, error) {
-	ch := make(chan *v1.Pod)
+type watcherFactory func(namespace string) watcher
+type watcher interface {
+	Watch(options metav1.ListOptions) (watch.Interface, error)
+}
 
+func (kCli K8sClient) makeWatcher(f watcherFactory, ls labels.Selector) (watch.Interface, error) {
 	// passing "" gets us all namespaces
-	watcher, err := kCli.core.Pods("").Watch(metav1.ListOptions{LabelSelector: ls.String()})
-	if err != nil {
+	watcher, err := f("").Watch(metav1.ListOptions{LabelSelector: ls.String()})
```

Comment:
what benefit do we get from watching all namespaces rather than just the one in the config?

Reply:
good questions! Two reasons:
1) The config doesn't actually restrict what namespace you deploy to, the RBAC rules do that. The config is just a hint at what your RBAC rules might be.
2) The vast majority of users won't have namespaces in their configs - I don't expect that anyone at Windmill team does. You'd usually have it set up for you by a cluster-admin.

Reply:
I might describe it as "the config just specifies the default namespace to use when you run kubectl without --namespace"

> I don't expect that anyone at Windmill team does
FWIW, I found this alias last week and have found it to be a *huge* QOL improvement when using non-default namespaces `alias kcd='kubectl config set-context $(kubectl config current-context) --namespace '` (though I expect it to cause me problems at some point when I change namespaces and unintentionally cause tilt to start deploying to the new namespace)

Reply:
hmmm...what i tested this last week, AFAICT, this doesn't affect what namespace you deploy to when you use tilt? 

the namespaces are in the yaml. if you don't have a namespace in your yaml, it still uses the default namespace, regardless of what's in your kubectl context.

were you seeing something different?

Reply:
yep, if I run `tilt up snack`, then in another window run `kcd fission`, then edit snack, then tilt deploys snack to the fission namespace

Reply:
hmmmm...that's different than i was seeing...will play with it a bit more...